core/src/main/java/com/amplifyframework/core/Amplify.java

```diff
@@ -137,11 +137,16 @@ public static void configure(@NonNull final AmplifyConfiguration configuration,
                 }
             }
             if ((context.getApplicationInfo().flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0) {
-                // Start activity for developer menu
+                // Start activity for developer menu on device shake
                 Intent mainIntent = new Intent(context, DeveloperMenuActivity.class);
                 mainIntent.setAction(Intent.ACTION_MAIN);
                 mainIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-                context.startActivity(mainIntent);
+                ShakeDetector shakeDetector = new ShakeDetector(context, () -> {
+                    if (!DeveloperMenuActivity.isActivityStarted()) {
+                        context.startActivity(mainIntent);
+                    }
+                });
+                shakeDetector.startDetecting();
```

Comment:
In the interest of keeping the `configure(...)` method as lean as possible, can we state this block as:
```
if (isDebuggableApplication()) {
    enableDeveloperMenu();
}
```
And then put the details into helpers?